File: src/main/java/com/emes/HashedFile.java
package com.emes;

public record HashedFile(
    String path,
    String hash
) {

  private static final String SEPARATOR = ",";

  @Override
  public String toString() {
    return path + SEPARATOR + hash;
  }

  public static HashedFile fromString(String string) {
    var split = string.split(SEPARATOR);
    return new HashedFile(split[0], split[1]);
  }
}

File: src/main/kotlin/com/emes/HashedFile.kt
package com.emes

data class HashedFile(
    val path: String,
    val hash: String
) {
    override fun toString(): String {
        return path + SEPARATOR + hash
    }

    companion object {
        private const val SEPARATOR = ","

        fun fromString(string: String): HashedFile {
            val split = string.split(SEPARATOR)
            return HashedFile(split[0], split[1])
        }
    }
}